### PR TITLE
(REF) Schema - Correct boolean fields in various tables 

### PR DESCRIPTION
**Pull request by monishdeb**
Overview
----------------------------------------
Followup PR #23134

### CODE REVIEW

**Comment by monishdeb** [comment]
@colemanw @demeritcowboy after reviewing each table boolean column I have prepared this patch. Can you please review this PR?

File: CRM/Upgrade/Incremental/php/FiveFortyNine/ACL.bool.php [hunk 1]
```diff
@@ -1,4 +1,6 @@
 <?php
 return [
-  // DAO changed, but nothing in upgrader?
+  'civicrm_acl_entity_role' => [
+    'is_active' => "DEFAULT 1 COMMENT 'Is this property active?",
```
Comment (demeritcowboy) [inline CRM/Upgrade/Incremental/php/FiveFortyNine/ACL.bool.php:4]: The test fail is because there's a single-quote missing near the end here.

**Comment by demeritcowboy** [comment]
I haven't `r-run` yet but I also see this, where there's conflicting defaults: https://github.com/civicrm/civicrm-core/blob/fdcb13f87f685985a48ebf434623517e64070b2c/xml/schema/Event/ParticipantStatusType.xml#L73-L75

And then it doesn't cause a problem but both of these fields have duplicate defaults: https://github.com/civicrm/civicrm-core/blob/fdcb13f87f685985a48ebf434623517e64070b2c/xml/schema/Financial/PaymentProcessorType.xml#L54-L68

**Comment by demeritcowboy** [comment]
Thanks @monishdeb. I think there's still a problem somewhere - on upgrade all the civicrm_menu.is_public entries get set to 1. I'll try to debug.

And also sorry just noticing this should be against 5.49.

**Comment by demeritcowboy** [comment]
Ok this line should be removed: https://github.com/civicrm/civicrm-core/blob/379de8beafc900626b9974910702f763a1094440/CRM/Upgrade/Incremental/php/FiveFortyNine/Core.bool.php#L62

**Comment by eileenmcnaughton** [comment]
@monishdeb I've updated the base branch to 5.49 so we don't merge to master - obviously it has a lot to rebase out now...

**Comment by eileenmcnaughton** [comment]
@monishdeb I've put https://github.com/civicrm/civicrm-core/pull/2319 into a separate PR - as it seems to e causing much confusion - this is obviously needing rebase but we can maybe merge that first -

**Comment by monishdeb** [comment]
@eileenmcnaughton I have rebased the PR. Also, should we also remove the requiredness of menu.is_public ?

**Comment by eileenmcnaughton** [comment]
@monishdeb looks better!

I think it is correct for booleans to be required - as long as we also have a default & do the upgrade script thing

**Comment by monishdeb** [comment]
ok reverting the Menu stuff.

**Comment by monishdeb** [comment]
Done. @eileenmcnaughton can you please review the PR?

**Comment by eileenmcnaughton** [comment]
OK - I took a look at what is in this patch & all the changes make sense - I haven't verified that there is complete cover of all the changes that need to be made - only that what is in this patch looks right

**Comment by monishdeb** [comment]
Thanks @eileenmcnaughton !

**Pull request merged**